### PR TITLE
Fix: Prevent Placeholder image import to media library

### LINKED ISSUE #17932: 🐞 Bug report: Placeholder image duplication on import [ Third Party Addon Pack issue ]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

The issue is not specific to elementor core template export-import but occurs with third-party addon packs. 

When they try to "X-copy/paste", "Cross-Domain Copy-Paste" elementor process data using `Elementor\Control_Media()->on_import()`

Here comes the main issue. When on settings has a placeholder image elementor treats that as an actual images and imports to the media library. 

For example:
```
(
    [id] => 
    [url] => http://my-remote.website/wp-content/plugins/elementor/assets/images/placeholder.png
)
```
this gets converted into this
```
(
    [id] => 3123
    [url] => http://my-other.website/wp-content/uploads/2022/02/placeholder-1.png
)
```
where `http://my-remote.website` is the source for copy data and `http://my-other.website` is the destination. 

This causes a lot of duplicate placeholder images in the media library.  This is not the only problem. In some cases where a lot of placeholder image exists like repeater with responsive media control, the import shows `504 server time out`.

### Steps to reproduce

In order to reproduce you need to install one of these plugins that allow Cross-domain copy/pasting. 

1. Happy Addons Pro (x-copy -> x-pase)
2. Elements Kit (Ecopy -> Epaste)
3. There are a few more plugin that has the same feature.

Steps to reproduce:
1. Create a section with multiple widgets where repeater should exist. 
2. Let the placeholder images as it is. 
3. copy the section with one of the previously mentioned plugins
4. paste it in a different WordPress installation. 
5. If you don't get 504 then you will find a media library full of placeholder images.

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

					
== Server Environment ==
	Operating System: Darwin
	Software: nginx/1.21.4
	MySQL version: Homebrew v8.0.27
	PHP Version: 8.1.0
	PHP Max Input Vars: 1000
	PHP Max Post Size: 512M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.9
	Site URL: http://ha.test
	Home URL: http://ha.test
	WP Multisite: No
	Max Upload Size: 512 MB
	Memory limit: 512M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: ha@local.test
	Debug Mode: Active

== Theme ==
	Name: Hello Elementor
	Version: 2.5.0
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36

== Active Plugins ==
	Dynamic Content for Elementor
		Version: 1.8.0
		Author: Dynamic.ooo

	Elementor
		Version: 3.5.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.6.2
		Author: Elementor.com

	Happy Elementor Addons
		Version: 3.4.3
		Author: weDevs

	Happy Elementor Addons Pro
		Version: 2.2.4
		Author: weDevs

== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Import Export Template Kit: Active by default
	Additional Custom Breakpoints: Active by default
	Hide native WordPress widgets from search results: Active by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Landing Pages: Active by default
	Color Sampler: Active by default
	Favorite Widgets: Active by default
	Admin Top Bar: Active by default
	Page Transitions: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default

== Log ==
	
PHP: showing 3 of 3PHP: 2022-02-15 11:11:29 [notice X 1080][/Users/sabbir/Web/office/ha/wp-content/plugins/elementor/core/upgrade/manager.php::62] version_compare(): Passing null to parameter #2 ($version2) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-15 12:40:57 [notice X 2][/Users/sabbir/Web/office/ha/wp-content/plugins/elementor/core/logger/items/base.php::46] Return type of Elementor\Core\Logger\Items\Base::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-22 14:11:57 [notice X 8][/Users/sabbir/Web/office/ha/wp-content/plugins/elementor/includes/controls/groups/base.php::497] str_replace(): Passing null to parameter #2 ($replace) of type array|string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2022-02-15 12:40:56 [error X 1][http://ha.test/wp-content/plugins/widget-builder-for-elementor/assets/js/editor.min.js?ver=1.0.3.1:2:3409] Cannot read properties of undefined (reading \'type\') 
JS: 2022-02-22 10:46:37 [error X 1][http://ha.test/wp-content/plugins/happy-elementor-addons-pro/assets/admin/js/marvin-new.min.js?ver=2.2.4:1:2762] widget is not defined 

== Elementor - Compatibility Tag ==
	
	Dynamic Content for Elementor: Compatibility not specified
	Elementor Pro: Compatible
	Happy Elementor Addons: Compatible
	Happy Elementor Addons Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	
	Happy Elementor Addons: Compatible
	Happy Elementor Addons Pro: Compatible

### PR DESCRIPTION
**Pull request by sh-sabbir**
added check for placeholder image. if URL contains images/placeholder.png then don't import image to the media library

## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Prevent duplicate placeholder images while importing templates. 

## Description
An explanation of what is done in this PR

* Fixed #17932

## Test instructions
This PR can be tested by following these steps:

* I can't write unit test

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [x] Docs have been added / updated (for bug fixes / features)

Fixes #17932

### CODE REVIEW

File: includes/controls/media.php [hunk 1]
```diff
@@ -64,11 +64,11 @@ public function get_default_value() {
 	 * @return array Control settings.
 	 */
 	public function on_import( $settings ) {
-		if ( empty( $settings['url'] ) ) {
+		if ( empty( $settings['url'] ) || strpos( $settings['url'], 'images/placeholder.png' ) !== false ) {
```
Comment (matipojo) [inline includes/controls/media.php:67]: Hi, thanks for the suggestion!

Can you please move the check to the `import` method that is responsible for the import?
`Plugin::$instance->templates_manager->get_import_images_instance()->import( $settings );`

Also, the check need to be more specific `elementor/assets/images/placeholder.png`.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/controls/media.php (hunk 1 above)
Comment (sh-sabbir) [reply includes/controls/media.php:67]: Thanks for your observation. I'll make the changes and commit. After that I'll let you know here.

Comment (sh-sabbir) [reply includes/controls/media.php]: I've adopted the changes requested.

**Comment by sh-sabbir** [comment]
Any update here?

**Comment by rami-elementor** [comment]
What if the placeholder image was changed using the `lementor/utils/get_placeholder_image_src` filter?

Ref: https://developers.elementor.com/docs/hooks/placeholder-image/

**Comment by sh-sabbir** [comment]
Update to target release/3.13.0

**Comment by sh-sabbir** [comment]
> rami-elementor

I'm here trying to solve the default placeholder issue which occurs 100% time. Now about your concern, Say for example we are importing from `Site A` to `Site B`. Now `Site A` has a custom placeholder defined using the `elementor/utils/get_placeholder_image_src` filter. While importing on `Site B` there is no way to determine that. So this is out of the scope of this PR. And I don't think that is possible to fix unless the default placeholder URL is defined somewhere in the export file.

**Comment by sh-sabbir** [comment]
@matipojo @KingYes 
Please check this PR for the latest updates https://github.com/elementor/elementor/pull/21378

**Pull request closed without merging**